pkg/controllers/provisioningv2/rke2/provisioningcluster/template.go

```diff
@@ -207,6 +208,19 @@ func toMachineTemplate(machinePoolName string, cluster *rancherv1.Cluster, machi
 	return ustr, nil
 }
 
+func populateHostnameLengthLimitAnnotation(mp rancherv1.RKEMachinePool, defaults rancherv1.RKEMachinePoolDefaults, annotations map[string]string) {
+	hostnameLimit := 0
+	if hl := mp.HostnameLengthLimit; hl >= rke2.MinimumHostnameLengthLimit && hl <= rke2.MaximumHostnameLengthLimit {
+		hostnameLimit = hl
```

Comment:
nit: should we log an error if it's not within the min/max? Otherwise the user may not understand why their arbitrary limit is not taking effect.